### PR TITLE
Story 16/17

### PR DESCRIPTION
**Pull request by katiekeel**
Authentication checks added for when user changes /orders/:id url, or when user is not logged in and tries to visit /orders. 

Also in this PR: test crash fixes from my previous PR, and an update to story 14 functionality after the Slack discussion with Mike.

What do you guys think of the index route in OrdersController? I could push it down to another method like Ellen did for the show view.

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -12,6 +16,9 @@ def show
   private
 
   def find_order
-    @order = Order.find(params[:id])
+    @order = current_user.orders.find(params[:id])
+    rescue ActiveRecord::RecordNotFound
+      flash[:notice] = "Please login appropriately to view that order."
```
Comment (emcooper) [inline app/controllers/orders_controller.rb]: Interesting, so the rescue basically acts as an if statement?

File: spec/features/authenticated/user_cannot_see_other_user_data_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    expect(page).to_not have_link("order-#{order_2.id}")
+
+    visit "/orders/#{order_2.id}"
+
+    expect(current_path).to eq "/login"
+    expect(page).to have_content "Please login appropriately to view that order."
+
+    visit '/admin'
+
+    expect(page).to have_button "Log In"
```
Comment (emcooper) [inline spec/features/authenticated/user_cannot_see_other_user_data_spec.rb]: They are already logged in at this point, right? Does visiting the admin page log them out?

Comment (emcooper) [inline spec/features/authenticated/user_cannot_see_other_user_data_spec.rb]: Oh, I guess b/c we're temporarily setting '/admin' to the login route. We'll just remember to change it later.

**Comment by emcooper** [comment]
Looks great. I think the index controller looks good. For the show I just always do that assuming that we'll need to 'find' for other actions as well.

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (katiekeel) [inline app/controllers/orders_controller.rb]: Yeah, the syntax is weird but that's essentially what it does. Originally I had it as an if @order.nil? thing but it was erroring out on the first line. The rescue saves it from the error.

File: spec/features/authenticated/user_cannot_see_other_user_data_spec.rb (hunk 2 above)
Comment (katiekeel) [inline spec/features/authenticated/user_cannot_see_other_user_data_spec.rb]: Right, yep. They're still logged in as a regular user but they would need to re-log in as an admin if they wanted to see admin stuff.

**Pull request merged**